app/components/rdf-input-fields/instance-multi-selector.js

```diff
@@ -0,0 +1,143 @@
+import { action } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+import { tracked } from '@glimmer/tracking';
+import {
+  triplesForPath,
+  updateSimpleFormValue,
+} from '@lblod/submission-form-helpers';
+import InputFieldComponent from '@lblod/ember-submission-form-fields/components/rdf-input-fields/input-field';
+import { restartableTask, timeout } from 'ember-concurrency';
+import { NamedNode, Namespace } from 'rdflib';
+
+export default class RdfInstanceMultiSelectorComponent extends InputFieldComponent {
+  inputId = 'select-' + guidFor(this);
+
+  @tracked selected = null;
+  @tracked options = [];
+  @tracked searchEnabled = true;
+
+  get subset() {
+    return this.options.slice(0, 50);
+  }
+
+  constructor() {
+    super(...arguments);
+    this.load();
+  }
+
+  async load() {
+    await this.loadOptions();
+    await this.loadProvidedValue();
+  }
+
+  async loadOptions() {
+    const instanceLabelProperty = this.getFormProperty('instanceLabelProperty');
+    const instanceApiUrl = this.getFormProperty('instanceApiUrl');
+
+    const pageSize = 5;
+    const response = await fetch(`${instanceApiUrl}?page[size]=${pageSize}`, {
+      headers: {
+        Accept: 'application/vnd.api+json',
+      },
+    });
+
+    this.options = await this.parseResponse(response, instanceLabelProperty);
+  }
+
+  async loadProvidedValue() {
+    if (this.isValid) {
+      const matches = triplesForPath(this.storeOptions, true).values;
+      let options = await Promise.all(
+        matches.map(async (m) => {
+          return await this.uriSearch(m.value);
+        })
+      );
+      this.selected = options
+        .map((m) => m[0])
+        .filter((opt) => matches.find((m) => m.equals(opt.subject)));
+    }
+  }
+
+  @action
+  updateSelection(options) {
```

Comment:
suggestion: rename to newSelection to make the difference with this.options more clear?